### DISCUSSION TITLE
`(sepPos != -1)` rather than `if (sepPos)`.

### DISCUSSION THREAD

**Discussion by VigilanteSystems** [jcassel/OctoPrint-Siocontrol#11] [Ideas]
Hi there, thanks for the plugin, nice work!
i had some troubles working your firmware, just had troubles on my side of development using a own "breed" of IO hardware, mine has parallel port inputs from a PC, anyway.. while debugging my crap i found in 
checkSerial() the following `if (sepPos)` statement. 
Here is some chatGPT reply to it: 
;) .. is it correct or not?maybe you want exactly the behaviour we see with (sepPos) only ... 👍 

*** ChatGPT ***
""
Based on the context and purpose of the code snippet, the correct variant to use would be `(sepPos != -1)` rather than `if (sepPos)`.

The reason for this is that the `indexOf()` function used earlier in the code returns -1 if the specified character or substring is not found in the given string. Therefore, using `if (sepPos)` as the condition would not accurately handle the case where the space character is not found.

By using `(sepPos != -1)` as the condition, the code will correctly handle the scenario where the space character is not present in the received input string. It ensures that the `command` variable is assigned the entire `buf` string in such cases.

To summarize, the code should be modified to use `(sepPos != -1)` instead of `if (sepPos)` to accurately check if the space character is found in the string. This ensures that the code behaves as intended in both scenarios, whether the space character is present or not.
""
*** ChatGPT ***

BUT it doesnt matter much, its a if else statement, and the else part is just debug messaging.
it gave me some more "fixes" which i still try out..

**Comment by jcassel** [comment]
What is the issue you are trying to solve?

**Comment by jcassel** [comment]
This might cause an issue in 3 commands. 
if you issue a debug command without a space or value, it will result in turning off debug.
If you issue an IO command without a value, with or without a space, it will set IO point #2 to the LOW State. 
If you issue an SE command without a value it will likely evaluate to a Zero and disable SE and and that might not be your intention. 

The only one of any concern here is the IO command.  If you have something sending garbage as described above, you might have an IO point change unexpectedly. Not good.  But also not a disaster and should not be an issue that shows up in the SIOControl PlugIn. 

The real issue here is that the 3 commands should have more code to evaluate the "Value" sent to them.. changing the if statement around should the command be parsed or not is something that should be done but it really does not solve the issue above where the space is included yet the value is not. Or another case where the value is not in the expected format / range. 

I likely will make the change to correct the IO command quickly since it affects IO. The rest can wait.

**Comment by jcassel** [comment]
@VigilanteSystems 
I made updates to all firmware versions to incorporate the discovered issue.  Main change is in parsing and checking Values. 

I am closing the topic.  

If you have further topics or questions related to the firmware, please post in the firmware discussion space.  I think most will be able to make the connection if reading this but it feels messy having this here and not in the firmware discussion area.
